pkg/gather/tasks_processing.go

```diff
@@ -0,0 +1,73 @@
+package gather
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/gather/common"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// TODO: docs
+
+type Task struct {
+	Name    string
+	F       common.GatheringClosure
+	CanFail bool
+}
+
+type GatheringFunctionResult struct {
+	FunctionName string
+	Records      []record.Record
+	Errs         []error
+	Panic        interface{}
+	TimeElapsed  time.Duration
+	IgnoreErrors bool
+}
+
+func HandleTasksConcurrently(ctx context.Context, tasks []Task) chan GatheringFunctionResult {
+	// TODO: consider using tasks pool with limited number of tasks run at the same time
+	// so that memory usage doesn't grow linearly with the number of added gatherers
+
+	resultsChan := make(chan GatheringFunctionResult)
+
+	go func() {
+		var wg sync.WaitGroup
+
+		for _, task := range tasks {
+			wg.Add(1)
+			go func(task Task) {
+				startTime := time.Now()
+				var panicked interface{}
+				var records []record.Record
+				var errs []error
+
+				func() {
```

Comment:
Do we need this wrapping function?

Reply:
yup, to catch the panic by `defer` and continue execution

Reply:
Ahh now I guess I see it. We want to call defer and get possible panicked result before writing to the channel.